### PR TITLE
Add blog post about optimizations in Erlang/OTP 27

### PR DESCRIPTION
**Pull request by bjorng**

### CODE REVIEW

File: _posts/2024-04-12-optimizations.md [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+signed integer that does not fit in 60 bits and therefore have to be
+stored on the heap with `rcx` containing a tagged pointer to the bignum on
+the heap.
+
+If `rcx` is a pointer to a term on the heap, it is not always safe to
+directly updating the existing tuple. That is because of the way the
+Erlang [generational garbage
+collector](https://www.erlang.org/doc/apps/erts/garbagecollection#generational-garbage-collection)
+works. Each Erlang process has two heaps for keeping Erlang terms:
+the young heap and the old heap. Terms on the young heap is allowed to
```
Comment (jhogberg) [inline _posts/2024-04-12-optimizations.md:336]: Nitpick:

```suggestion
the young heap and the old heap. Terms on the young heap are allowed to
```

File: _posts/2024-04-12-optimizations.md [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+    mov rdi, rcx           ; Get new value for atoms field
+    mov qword ptr [rax+22], rdi
+    mov qword ptr [rbx+8], rax
+```
+
+[avx]: https://en.wikipedia.org/wiki/Advanced_Vector_Extensions
+
+If all the new values being written into the existing record are known
+never to be tagged pointers, the native instructions can be
+simplified. Consider this module:
```
Comment (jhogberg) [inline _posts/2024-04-12-optimizations.md:384]: Should we mention that it's safe for literals too?

File: _posts/2024-04-12-optimizations.md [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+madd(A, C) ->
+    fun(B) -> A * B + C end.
+```
+
+In Erlang/OTP 26, the native code for creating the fun looks like so:
+
+```
+# i_make_fun3_FStt
+L38:
+    long mov rsi, 9223372036854775807
```
Comment (jhogberg) [inline _posts/2024-04-12-optimizations.md:479]: `9223372036854775807` is a placeholder value to force a long move, which might get confusing further down when we have several values in the same example. Perhaps we should explain what the values are? e.g `; fun dispatch table`

File: _posts/2024-04-12-optimizations.md [hunk 4]
```diff
@@ -0,0 +1,771 @@
...
+  faster than the Erlang version.)
+
+* The runtime system's function for doing multiplication of large
+  integers was modified to use the [Karatsuba
+  algorithm](https://en.wikipedia.org/wiki/Karatsuba_algorithm), which
+  is a divide-and-conquer multiplication algorithm invented in the 1960s.
+
+* Some of the low-level helper functions for arithmetic with large
+  integers (bignums) were modified to take advantage of a 128-bit
+  integer data type on 64-bit CPUs (if supported by the C compiler).
```
Comment (jhogberg) [inline _posts/2024-04-12-optimizations.md:671]: Perhaps skip the parenthesis?
```suggestion
  integer data type on 64-bit CPUs when supported by the C compiler.
```

File: _posts/2024-04-12-optimizations.md (hunk 2 above)
Comment (bjorng) [reply _posts/2024-04-12-optimizations.md:384]: Yes, good idea. I will make a new example that updates a record with a literal fun.

File: _posts/2024-04-12-optimizations.md (hunk 3 above)
Comment (bjorng) [reply _posts/2024-04-12-optimizations.md:479]: Will do.

**Pull request merged**